lib/view_component/preview.rb

```diff
@@ -24,7 +24,7 @@ def render_args(example, params: {})
         provided_params = params.slice(*example_params_names).to_h.symbolize_keys
         result = provided_params.empty? ? new.public_send(example) : new.public_send(example, **provided_params)
         @layout = nil unless defined?(@layout)
-        result.merge(layout: @layout)
+        { layout: @layout }.tap { |args| args.merge!(result) if result }
```

Comment:
This implementation feels unexpected. `tap` to call `merge!` because it as side effects is hard to read and argue about. Why do we need to handle the case where `result` is empty?

Reply:
Because result can be actually nil, instead of empty. 

I guess we can rewrite this as:
```suggestion
        result ||= {}
        result.merge(layout: @layout)
```

Reply:
I'm trying to understand how result can be empty. Does your change introduce that case or has it always been like this? If the former can't we fix it if the later how come it hasn't come up before?

Reply:
My change introduce that case, because the example method is now empty, it's just the mere definition of the example method:
```ruby
class MyComponentPreview < ViewComponent::Preview
  def default # body-less method
  end
end
```

Hence, this is why the result can be `nil`, because `new.public_send` will always return nothing (in these cases where a template is used)

I guess we can refactor this still a bit further and avoid calling `new.public_send` when using a template no? And maybe even pass the template path as part of the return of this `render_args` method? Wdyt?